### ISSUE TITLE
Allow use of app without connected wallet/"demo mode"

### ISSUE THREAD

**Issue by 0xdef1cafe** [shapeshift/web#895]
# Overview

Many users have requested to be able to use the app (app.shapeshift.com) without first connecting a wallet. This makes sense from an engagement perspective - people want to see what things look and feel like before they use it.

# Reference

- instantiate a demo wallet like shown https://github.com/shapeshift/hdwallet/pull/452/commits/e59e755b6403cdfd2bbb331df2a0f535eef00a31#diff-673a1350d3a18db303ae7b7494c175c13e74e0dbd8faef3b1200d9708b028f9aR334-R335
- make an initial PR to update the `useWallet` hooks to provide an `isDemoWallet` boolean

# Guidance

At a high level, the app should be as functional as possible, without the user being able to get into a broken state.

- Use the `useWallet` hook to detect the presence (or rather, lack thereof) of a wallet for logic
- Add a grey button to the splash screen below the Connect Wallet, saying "Try without a wallet..."
- Redirect to [Bitcoin asset page](https://app.shapeshift.com/assets/bitcoin) as the first screen instead of `/dashboard`
- Disable send/receive buttons on asset pages
- Hide account page link in nav
- Hide accounts from asset pages
- Hide balance chart buttons, make sure pages default to price charts
- Disconnect wallet button should disconnect wallet and clear portfolio, but not boot the user back out to the `connect-wallet` screen
- Trade dialog
  - Sell assets should be fetched from all assets list, not users portfolio assets
  - Confirm trade button should be disabled
- Defi vaults - show modal but stop the user at step 1

# Acceptance criteria

- No regressions across the app
- App is usable to the fullest extent possible without connecting a wallet

# Bounty Hunters 
- Join our discord [here](https://discord.gg/shapeshift). Once you join you can see more about this issue [here](https://discord.com/channels/554694662431178782/873244924026449940/925799058537336853)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 5750.0084 FOX (2097.06 USD @ $0.36/FOX)  attached to it as part of the ShapeShift fund.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/shapeshift/web/895/100027568).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/shapeshift/web/895/100027568).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $4,730,926.00 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by amitojsingh366** [comment]
Could I attempt this bounty, I feel like I will be able to achieve the desired result here

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 264 years, 9 months from now.
Please review their action plans below:


**1) [amitojsingh366](https://gitcoin.co/amitojsingh366) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/shapeshift/web/895/100027568?mutate_worker_action=approve&worker=amitojsingh366) | [reject worker](https://gitcoin.co/issue/shapeshift/web/895/100027568?mutate_worker_action=reject&worker=amitojsingh366))_.**

I would love to work on this bounty, I already have a lot of experience with this code base

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/shapeshift/web/895/100027568).

**Comment by oscardavidrm** [comment]
In case the bounty assignee spot opens again, can I give it a try?

**Comment by amitojsingh366** [comment]
> In case the bounty assignee spot opens again, can I give it a try?

Hi @oscardavidrm I have already started work on this bounty as I got assigned to this issue, I think I will have this ready by the end of today :)

**Comment by gall0x** [comment]
Sorry, I didn't see someone was already working on it.

**Comment by cjthompson** [comment]
@0xdef1cafe - My personal preference would be to remove the splash screen entirely and default to using the app without a wallet. There is already a button in the upper-right to connect a wallet.

another option would be that rather than a full screen splash screen, a small modal could be shown over the app on first load.

I'm not a fan on the splash screen.

**Comment by stackedq** [comment]
@cjthompson with respect to UX perspective, I suggest removing splash screen. But opening up a modal on user arrival is really annoying, I believe the better workaround is to open the "connect wallet" modal after clicking wallet related CTA buttons.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__The funding of 5750.0084 FOX  (1954.43 USD @ $0.34/FOX) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $4,662,632.65 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
⚡️ A tip worth 1000.00000  FOX (337.63 USD @ $0.34/FOX) has been granted to @amitojsingh366 for this issue from @0xean. ⚡️ 

Nice work @amitojsingh366! Your tip has automatically been deposited in the ETH address we have on file.

 * $4662632.30 in Funded OSS Work Available at: https://gitcoin.co/explorer
 * Incentivize contributions to your repo: <a href='https://gitcoin.co/tip'>Send a Tip</a> or <a href='https://gitcoin.co/funding/new'>Fund a PR</a>
 * No Email? Get help on the <a href='https://discord.gg/gitcoin'>Gitcoin's Discord</a>

**Comment by 0xdef1cafe** [comment]
awaiting mockups from product and can be rebountied as "demo mode"

**Comment by 0xean** [comment]
Product to meet with @cjthompson  on some wallet functionality before we move forward.

**Comment by mrnerdhair** [comment]
The associated hdwallet PR is shapeshift/hdwallet#452 for reference

**Comment by 0xean** [comment]
Pending a HD wallet release, this is ready for work to begin in web. @0xdef1cafe - to bounty out.

**Comment by 0xean** [comment]
@0xf11nix - any interest in this one? We can get a bounty for you if so

**Comment by 0xf11nix** [comment]
> @0xf11nix - any interest in this one? We can get a bounty for you if so

@0xean. Sure, I'd love to take a crack at this one

**Comment by LPX55** [comment]
Can we get a status update on this? Translation team has requested this feature to more easily access certain areas of the platform. @0xf11nix

**Comment by 0xf11nix** [comment]
GM @LPX55!

My apologies for the delay in getting back to you on this one. We did have a couple of hiccups along the way relating to the spec/scope of the changes, but this has since been resolved.

Regarding an update on a timeline for release; I have now opened a PR (#1585), which is still in draft (pending a couple more changes before we can rollout to production), but we are on track for this being ready for a final review in the coming days.

I have been working with both Diggy and Lych over the last week or so, and will reach out to you personally once we have an ETA on next steps, and a timeline for rollout to production.

**Comment by gomesalexandre** [comment]
> As a user, It feels a bit weird having to switch from the demo wallet to another wallet, I would probably expect to have a "Login" button instead of the wallet dropdown inside the navbar on the top right, reducing the wallet modal access to one action instead of 2, as disconnecting from the demo wallet makes no sense

Agreed - a "Connect Wallet" button instead of the usual wallet dropdown would be a better UX for the demo wallet.

**Comment by 0xf11nix** [comment]
Closing in favour of https://github.com/shapeshift/web/issues/1621
